pkg/frontend/frontend.go

```diff
@@ -320,23 +317,23 @@ func (f *frontend) chiAuthenticatedRoutes(router chi.Router) {
 
 				r.Get("/clusterdeployment", f.getAdminHiveClusterDeployment)
 
-				r.Post("/redeployvm", f.postAdminOpenShiftClusterRedeployVM)
+				r.With(f.maintenanceMiddleware.UnplannedMaintenanceSignal).Post("/redeployvm", f.postAdminOpenShiftClusterRedeployVM)
 
-				r.Post("/stopvm", f.postAdminOpenShiftClusterStopVM)
+				r.With(f.maintenanceMiddleware.UnplannedMaintenanceSignal).Post("/stopvm", f.postAdminOpenShiftClusterStopVM)
 
-				r.Post("/startvm", f.postAdminOpenShiftClusterStartVM)
+				r.With(f.maintenanceMiddleware.UnplannedMaintenanceSignal).Post("/startvm", f.postAdminOpenShiftClusterStartVM)
 
-				r.Post("/upgrade", f.postAdminOpenShiftUpgrade)
+				r.With(f.maintenanceMiddleware.UnplannedMaintenanceSignal).Post("/upgrade", f.postAdminOpenShiftUpgrade)
 
 				r.Get("/skus", f.getAdminOpenShiftClusterVMResizeOptions)
 
-				r.Post("/resize", f.postAdminOpenShiftClusterVMResize)
+				r.With(f.maintenanceMiddleware.UnplannedMaintenanceSignal).Post("/resize", f.postAdminOpenShiftClusterVMResize)
```

Comment:
My only suggestion would be to not emit an unplanned maintenance signal for the resize endpoint, since this is always something we come to agreement on with the customer before performing. This might change in the future, but for now the resize is planned maintenance only.